For_Andrey/task_3.py

```diff
@@ -0,0 +1,175 @@
+import random
+from math import ceil
+# Использовал math в методе number_of_car_refueling для округления до болшего числа. Можно было и без него.
+# Если модули лишний раз лучше не импортировать - дай знать.
+
+
+class Car:
+
+    New_car_cost = 10000
+    Repair_for_diesel_car = 700
+    Repair_for_petrol_car = 500
+    Mileage_before_repair_diesel = 150_000
+    Mileage_before_repair_petrol = 100_000
+    Diesel_fuel_cost = 1.8
+    Petrol_fuel_cost = 2.4
+    Fuel_consumption_diesel_on_hundred_km = 6 / 100
+    Fuel_consumption_petrol_on_hundred_km = 8 / 100
+    Decrease_value_diesel_per_thousand_km = 105
+    Decrease_value_petrol_per_thousand_km = 95
+
+    def __init__(self, engine, gas_tank):
+        assert engine in ('diesel', 'petrol'), f'Incorrect engine type: {engine}'
+        assert gas_tank in (60, 75), f'Incorrect gas_tank type: {gas_tank}'
+        self.engine = engine
+        self.gas_tank = gas_tank
+        self.mileage = random.randint(55_000, 286_000)
+
+    # Пробег
+    def car_mileage(self):
+        mileage = self.mileage
+        return mileage
+
+    # Рассчитываем количество тех.обслуживаний машины
+    def number_of_repairs(self):
+        mileage = self.mileage
+        number_of_repairs = 0
+        if self.engine == 'diesel':
+            number_of_repairs = mileage // Car.Mileage_before_repair_diesel
+        elif self.engine == 'petrol':
+            number_of_repairs = mileage // Car.Mileage_before_repair_petrol
+        return number_of_repairs
+
+    # Рассчитываем стоимость всех тех.обслуживаний
+    def repair_cost(self):
+        number_of_repairs = Car.number_of_repairs(self)
+        repair_cost = 0
+        if self.engine == 'diesel':
+            repair_cost = Car.Repair_for_diesel_car * number_of_repairs
+        elif self.engine == 'petrol':
+            repair_cost = Car.Repair_for_petrol_car * number_of_repairs
+        return repair_cost
+
+    # Получился очень длинный метод для рассчёта стоимости бензина. Короче не придумал
+    # Главная проблема в том, что расход увеличивается на 1% с каждой 1000 км. Поэтому много букв
+    # В конце округлил до ближайшего целого числа
+    def fuel_cost(self):
```

Comment:
This method should be rewritten, you have used incorrect logic.
I would not say that https://github.com/akopylov89/Class_work-P5-/blob/master/Classes.py
is a correct interpretation, but something could be taken from there

Reply:
This is an absolutely ingenious method. Method ready for anything. I can prove it when calling :)